es-app/src/FileData.cpp

```diff
@@ -134,12 +126,27 @@ std::vector<FileData*> FileData::getFilesRecursive(unsigned int typeMask, bool d
 	return out;
 }
 
+std::string FileData::getKey() {
+	if (mIsVirtual)
+		return getLongKey();
+	else
+		return getShortKey();
+}
+
+FileData* FileData::getMetadataSourceFileData()
```

Comment:
Hm. We will return the actual file data, or the source file data (i.e. the original filedata of which the current instance is a clone of). Would "get()" not be misleading? Happy to reconsider the name, though - it's just that "get()" might not be accurate.